src/cc/api/BPF.cc

```diff
@@ -787,14 +792,22 @@ bool USDT::operator==(const USDT& other) const {
          (probe_func_ == other.probe_func_);
 }
 
+int USDT::set_probe_matching_kludge(uint8_t kludge) {
```

Comment:
this function is not used?

Reply:
This is intended to be called by the program after creating a `USDT` object, before the usdt object(s) are passed to `BPF::init`. This way all `USDT::Context`s and `USDT::Probe`s created by the USDT have their `mod_match_inode_only_` members set and the "match inode only, ignore dev" kludge is successfully enabled.

For example, my program which runs on btrfs hosts does the following:
```cpp
std::vector<USDT> usdts;
auto u = USDT(
      pid,
      Provider.str(),
      Name.str(),
      ProbeFuncName);
u.set_probe_matching_kludge(1);
usdts.push_back(u);
bpf_.init(bpfProgramToRun, cflags, usdts);
```

I debated whether it made sense to put this set-kludge option in the `USDT` constructor but decided against it because:
  * Hopefully sometime in future btrfs fixes device id disparity between `stat` and `procfs` and I can remove this logic.
  * This is sufficiently obscure of a corner case that this kludgey setter function felt appropriate versus adding another constructor arg

I don't feel particularly strong about setter vs constructor thing here, so happy to change if you'd like.

Reply:
Then could you add some comments for the function. Since this will be part of API, we need to document when to use it.

Reply:
comments added